src/Symfony/Component/Mime/Part/TextPart.php

```diff
@@ -207,7 +208,18 @@ private function getEncoder(): ContentEncoderInterface
             return self::$encoders[$this->encoding] ??= new QpContentEncoder();
         }
 
-        return self::$encoders[$this->encoding] ??= new Base64ContentEncoder();
+        if ('base64' === $this->encoding) {
+            return self::$encoders[$this->encoding] ??= new Base64ContentEncoder();
+        }
+
+        return self::$encoders[$this->encoding];
+    }
+
+    public function addEncoder(string $name, ContentEncoderInterface $encoder): static
+    {
+        self::$encoders[$name] = $encoder;
+
+        return $this;
     }
```

Comment:
```suggestion
    final public static function addEncoder(string $name, ContentEncoderInterface $encoder): void
    {
        self::$encoders[$name] = $encoder;
    }
```

The encoders array is static, so that mutator should be static as well. This way, we indicate that calling this method does not only affect a single `TextPart` instance but all instances globally.